### DISCUSSION TITLE
Progress Bar display

### DISCUSSION THREAD

**Discussion by llaville** [composer-unused/composer-unused#540] [Show and tell]
Hello,

For little cosmetic change, I suggest to `clear` the progress bar when job finished to avoid such results (Progress bar is still visible) on console !

![composer-unused-089_issue-1](https://github.com/composer-unused/composer-unused/assets/364342/2c99e3ee-fe4a-458b-9a42-be2859945300)
![composer-unused-089_issue-1b](https://github.com/composer-unused/composer-unused/assets/364342/13aeb019-0e79-4f7f-af1c-ad30fcb2dc14)
![composer-unused-089_issue-1c](https://github.com/composer-unused/composer-unused/assets/364342/47ee1f9b-d8b3-44af-9eeb-fd20fb5e10e9)

Quick change is to apply on https://github.com/composer-unused/composer-unused/blob/0.8.9/src/Console/Progress/DefaultProgressBarDecorator.php#L39-L44

```php
namespace ComposerUnused\ComposerUnused\Console\Progress;

class DefaultProgressBarDecorator implements ProgressBarInterface

    public function finish(): void
    {
        if ($this->isSetProgress()) {
            $this->progressBar->finish();
            $this->progressBar->clear();
        }
    }
```

And also removed the [`NullProgressBar`](https://github.com/composer-unused/composer-unused/blob/0.8.9/src/Console/Progress/NullProgressBar.php) that is not necessary : 
- see control by https://github.com/composer-unused/composer-unused/blob/0.8.9/src/Console/Progress/DefaultProgressBarDecorator.php#L46
- and https://github.com/composer-unused/composer-unused/blob/0.8.9/src/Console/Progress/DefaultProgressBarDecorator.php#L21-L22

So the final patch may be 

```
diff --git a/src/Console/Progress/DefaultProgressBarDecorator.php b/src/Console/Progress/DefaultProgressBarDecorator.php
index eebfd20..0cac53f 100644
--- a/src/Console/Progress/DefaultProgressBarDecorator.php
+++ b/src/Console/Progress/DefaultProgressBarDecorator.php
@@ -9,16 +9,13 @@ use Symfony\Component\Console\Output\OutputInterface;

 class DefaultProgressBarDecorator implements ProgressBarInterface
 {
-    /**
-     * @var NullProgressBar|ProgressBar
-     */
-    private $progressBar;
+    private ProgressBar $progressBar;

     private bool $noProgress;

     public function __construct(OutputInterface $io, int $max = 0, bool $noProgress = false)
     {
-        $this->progressBar = $noProgress ? new NullProgressBar() : new ProgressBar($io, $max);
+        $this->progressBar = new ProgressBar($io, $max);
         $this->noProgress = $noProgress;
     }

@@ -40,6 +37,7 @@ class DefaultProgressBarDecorator implements ProgressBarInterface
     {
         if ($this->isSetProgress()) {
             $this->progressBar->finish();
+            $this->progressBar->clear();
         }
     }

```

All checks PASSED 

```
> phpcs --parallel=50 --cache=data/cs-check.json
............................... 31 / 31 (100%)


Time: 186ms; Memory: 8MB

> phpstan analyse --no-progress
Note: Using configuration file /shared/backups/forks/composer-unused/phpstan.neon.


 [OK] No errors


> phpunit
PHPUnit 9.6.7 by Sebastian Bergmann and contributors.

..SS..................................................            54 / 54 (100%)

Time: 00:00.181, Memory: 42.00 MB

OK, but incomplete, skipped, or risky tests!
Tests: 54, Assertions: 102, Skipped: 2.
```

**Comment by icanhazstring** [comment]
Looks good so 👍
But removing the NullProgress, wouldn't that show a progress when the option to suppress it is active?

But could be that this was because of the log outputs which you say are gone now? 🤔

**Comment by llaville** [comment]
> But removing the NullProgress, wouldn't that show a progress when the option to suppress it is active?

No progress displayed even if you used the `--no-progress` option (already tested)

**Reply by icanhazstring** [reply]
Nice. Then it's fine 👍
Go ahead and file the PR. Happy to merge 🙂

**Reply by llaville** [reply]
PR #542 available
